plugins/modules/ipatrust.py

```diff
@@ -0,0 +1,282 @@
+  state:
+    description: State to ensure
+    default: present
+    required: true
+    choices: ["present", "absent"]
+
+version_added: "2.9"
+author:
+    - Rob Verduijn
+"""
+
+EXAMPLES = """
+# add ad-trust
+- ipatrust:
+    realm: example.com
+    trust_type: ad
+    realm_admin: Administrator
+    realm_passwd: Welcome2020!
+    state: present
+
+# delete ad-trust
+- ipatrust:
+    realm: example.com
+    state: absent
+"""
+
+RETURN = """
+"""
+
+
+if six.PY3:
+    unicode = str
+
+
+def find_trust(module, realm):
+    _args = {
+        "all": True,
+        "cn": realm,
+    }
+
+    _result = api_command(module, "trust_find", realm, _args)
+
+    if len(_result["result"]) > 1:
+        module.fail_json(msg="There is more than one realm '%s'" % (realm))
+    elif len(_result["result"]) == 1:
+        return _result["result"][0]
+    else:
+        return None
+
+
+def del_trust(module, realm):
+    _args = {}
+
+    _result = api_command(module, "trust_del", realm, _args)
+    if len(_result["result"]["failed"]) > 0:
+        module.fail_json(
+            msg="Trust deletion has failed for '%s'" % (realm))
+    else:
+        return None
+
+
+def add_trust(module, realm, args):
+    _args = args
+
+    _result = api_command(module, "trust_add", realm, _args)
+
+    if "cn" not in _result["result"]:
+        module.fail_json(
+            msg="Trust add has failed for '%s'" % (realm))
+    else:
+        return None
+
+
+def gen_args(trust_type, realm_admin, realm_passwd, realm_server, trust_secret, base_id, range_size, range_type, bidirectional, external):
+    _args = {}
+    if trust_type is not None:
+        _args["trust_type"] = trust_type
+    if realm_admin is not None:
+        _args["realm_admin"] = realm_admin
+    if realm_passwd is not None:
+        _args["realm_passwd"] = realm_passwd
+    if realm_server is not None:
+        _args["realm_server"] = realm_server
+    if trust_secret is not None:
+        _args["trust_secret"] = trust_secret
+    if base_id is not None:
+        _args["base_id"] = base_id
+    if range_size is not None:
+        _args["range_size"] = range_size
+    if bidirectional is not None:
+        _args["bidirectional"] = bidirectional
+    if external is not None:
+        _args["external"] = external
+
+    return _args
+
+
+def main():
+    ansible_module = AnsibleModule(
+        argument_spec=dict(
+            # general
+            ipaadmin_principal=dict(type="str", default="admin"),
+            ipaadmin_password=dict(type="str", required=False, no_log=True),
+            realm=dict(type="str", default=None, required=True),
+            # state
+            state=dict(type="str", default="present",
+                       choices=["present", "absent"]),
+            # present
+            trust_type=dict(type="str", default="ad", required=False),
+            admin=dict(type="str", default=None, required=False),
+            passwordd=dict(type="str", default=None,
```

Comment:
Is the second 'd' a typo? This parameter should match the CLI and be `password`.

Reply:
done